app/scripts/models/flow.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent a login flow, holding the data that we
+ * need to submit as part of our metrics when we perform actions
+ * during the flow.
+ *
+ * Try to fetch the flowId and flowBegin from the resume token.
+ * If unavailable there, fetch from data attributes in the DOM.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var Backbone = require('backbone');
+  var Cocktail = require('cocktail');
+  var ErrorUtils = require('lib/error-utils');
+  var ResumeTokenMixin = require('models/mixins/resume-token');
+  var SearchParamMixin = require('models/mixins/search-param');
+  var vat = require('lib/vat');
+
+  var Model = Backbone.Model.extend({
+    initialize: function (options) {
+      options = options || {};
+
+      this.sentryMetrics = options.sentryMetrics;
+      this.window = options.window || window;
+
+      this.fetch();
+    },
+
+    defaults: {
+      flowBegin: null,
+      flowId: null
+    },
+
+    fetch: function () {
+
+      var $body = this.window['$']('body');
+
+      this.populateFromStringifiedResumeToken(this.getSearchParam('resume'));
+
+      var attrsFromBodyData = {};
+      var error = null;
+
+      if (! this.get('flowId')) {
+        attrsFromBodyData.flowId = $body.data('flowId');
+        if (! attrsFromBodyData.flowId) {
+          error = AuthErrors.toMissingDataAttributeError('flowId');
+          ErrorUtils.captureError(error, this.sentryMetrics);
+        }
+      }
+
+      if (! this.get('flowBegin')) {
+        attrsFromBodyData.flowBegin = $body.data('flowBegin');
+        if (! attrsFromBodyData.flowBegin) {
+          error = AuthErrors.toMissingDataAttributeError('flowBegin');
+          ErrorUtils.captureError(error, this.sentryMetrics);
+        }
+      }
+
+      var valid = vat.validate(attrsFromBodyData, this.resumeTokenSchema);
```

Comment:
Validation should already have been done for you at this point, by `ResumeTokenMixin::populateFromResumeToken`.